flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -276,6 +281,7 @@ protected StreamTask(
 		this.actionExecutor = Preconditions.checkNotNull(actionExecutor);
 		this.mailboxProcessor = new MailboxProcessor(this::processInput, mailbox, actionExecutor);
 		this.mailboxProcessor.initMetric(environment.getMetricGroup());
+		this.mainMailboxExecutor = new MailboxExecutorImpl(mailbox, MIN_PRIORITY, actionExecutor);
```

Comment:
I was thinking about doing here:
```
this.mainMailboxExecutor = mailboxProcessor.getMainMailboxExecutor();
```
or
```
this.mainMailboxExecutor = mailboxProcessor.getMinPriorityMailboxExecutor();
```
to maintain the contract, that everything goes via `MailboxProcessor` one way or another.

As it is now, some actions are going via `mailbox` and `actionExecutor` bypassing the processor, which I'm not sure if it's as clean?

Reply:
Done.